codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/localservice/nameresolver/SmithyNameResolver.java

```diff
@@ -70,9 +75,18 @@ public static String smithyTypesNamespace(final Shape shape) {
       if (shape.hasTrait(ServiceTrait.class)) {
         return sdkName;
       }
-      return sdkName.concat("types");
+      // Boolean to hold if shape is input or output of any operation
+      boolean isTopLevelShape = model
+        .shapes(OperationShape.class)
+        .anyMatch(op ->
+          op.getInput().filter(shape.getId()::equals).isPresent() ||
+          op.getOutput().filter(shape.getId()::equals).isPresent()
+        );
+      return isTopLevelShape ? sdkName : sdkName.concat("types");
```

Comment:
Why not overload the method and pass this boolean only when required?

Reply:
Because if we do that every time we call `smithyTypesNamespace` we will end-up doing a if-else check like below:
```
if (isShapeFromAWSSDK(shape)) {
    smithyTypesNamespace(..., boolean flag);
} else {
    smithyTypesNamespace(...);
}
```